vtkext/private/module/F3DColoringInfoHandler.cxx

```diff
@@ -0,0 +1,193 @@
+        }
+      }
+
+      // Set component names
+      if (array->HasAComponentName())
+      {
+        for (size_t i = 0; i < static_cast<size_t>(array->GetNumberOfComponents()); i++)
+        {
+          const char* compName = array->GetComponentName(i);
+          if (i < info.ComponentNames.size())
+          {
+            if (compName && info.ComponentNames[i] != std::string(compName))
+            {
+              // set non-coherent component names to empty string
+              info.ComponentNames[i] = "";
+            }
+          }
+          else
+          {
+            // Add components names to the back of the component names vector
+            info.ComponentNames.emplace_back(compName ? compName : "");
+          }
+        }
+      }
+    }
+  }
+}
+
+//----------------------------------------------------------------------------
+std::optional<F3DColoringInfoHandler::ColoringInfo> F3DColoringInfoHandler::SetCurrentColoring(bool enable, bool useCellData, std::optional<std::string> arrayName, bool quiet)
+{
+  this->CurrentUsingCellData = useCellData;
+  auto& data =
+    this->CurrentUsingCellData ? this->CellDataColoringInfo : this->PointDataColoringInfo;
+  int nIndices = static_cast<int>(data.size());
+
+  if (!enable)
+  {
+    // Not coloring
+    this->CurrentColoringIter.reset();
+  }
+  else if (nIndices == 0)
+  {
+    // Trying to color but no array available
+    this->CurrentColoringIter.reset();
+
+    if (!quiet)
+    {
+      F3DLog::Print(F3DLog::Severity::Debug, "No array to color with");
+    }
+  }
+  else if (!arrayName.has_value())
+  {
+    // Coloring with first array
+    this->CurrentColoringIter = data.begin();
+  }
+  else
+  {
+    // Coloring with named array
+    this->CurrentColoringIter = data.find(arrayName.value());
+    if (this->CurrentColoringIter.value() == data.end())
+    {
+      // Could not find named array
+      this->CurrentColoringIter.reset();
+      if (!quiet)
+      {
+        F3DLog::Print(F3DLog::Severity::Warning, "Unknown scalar array: \"" + arrayName.value() + "\"\n");
+      }
+    }
+  }
+  return this->GetCurrentColoringInfo();
+}
+
+//----------------------------------------------------------------------------
+std::optional<F3DColoringInfoHandler::ColoringInfo> F3DColoringInfoHandler::GetCurrentColoringInfo() const
+{
+  if (this->CurrentColoringIter.has_value())
+  {
+    return this->CurrentColoringIter.value()->second;
+  }
+  else
+  {
+    return std::optional<ColoringInfo>();
+  }
+}
+
+//----------------------------------------------------------------------------
+void F3DColoringInfoHandler::CycleColoringArray(bool cycleToNonColoring)
+{
+  auto& data =
+    this->CurrentUsingCellData ? this->CellDataColoringInfo : this->PointDataColoringInfo;
+  if (!this->CurrentColoringIter.has_value())
+  {
+    if(data.empty())
+    {
+      return;
+    }
+    else
+    {
+      this->CurrentColoringIter = data.begin();
+    }
```

Comment:
either
```suggestion
    if(!data.empty())
    {
      this->CurrentColoringIter = data.begin();
    }
```
or
```suggestion
    if(data.empty())
    {
      return;
    }
    this->CurrentColoringIter = data.begin();
```
to avoid the mix of complete branching and early return?